### ISSUE TITLE
Add a command to check the status of a current match by discord user id

### ISSUE THREAD

**Issue by B3none** [csgo-league/csgo-league-bot#7]
It'll require an endpoint on the backend but I think it would be nice to add a command for this as people in the current queue might want to know how long they will be waiting for players in that match to join.

**Comment by cameronshinn** [comment]
In the API could this become an extension of `in_match`? Instead of having [this function](https://github.com/csgo-league/csgo-queue-bot/blob/5b78c30b2786d13f1511c81f2e7ced863204517f/qbot/helpers/api.py#L140) return a boolean, we can have it return the player's match *if* they are in one, otherwise return `None`. This way the return value can be treated like a boolean (and is functionally the same), but can also be used to get the match object.

EDIT: Oops I was thinking we had an `in_match` function, but the one I linked is `is_linked`. I realize we check if they are in a match from their profile object.

**Comment by B3none** [comment]
I could alter the existing inMatch response to return either false OR an object for the current the current map of their game if they’re in one. Would that help?

The only problem is that then the player response becomes quite bloated. I think it’d be easier to get a match by someone’s discord Id in a separate endpoint.

**Comment by cameronshinn** [comment]
Yeah I think having a separate endpoint would probably be best.

As a side note, in the player object we could replace `inMatch` with `current_match` which is either `null` or the match ID. More informative that way.

**Comment by WardPearce** [comment]
Could introduce a matches route what allows you to pull multiple matches based off your desired parameters, if no parameters given just return X amount of the latest matches.

This would allow you to find matches based off Discord IDs, Match IDs etc..

**Comment by cameronshinn** [comment]
> Could introduce a matches route what allows you to pull multiple matches based off your desired parameters, if no parameters given just return X amount of the latest matches.
> 
> This would allow you to find matches based off Discord IDs, Match IDs etc..

Yeah this could be a good all-in-one solution. Get an active match by player ID, steam ID, match ID.

**Comment by B3none** [comment]
Okay, I'll implement and endpoint on the web and let you know when it's merged.

**Comment by B3none** [comment]
> Could introduce a matches route what allows you to pull multiple matches based off your desired parameters, if no parameters given just return X amount of the latest matches.
> 
> This would allow you to find matches based off Discord IDs, Match IDs etc..

I would probably steer away from this because I don't want one `jack of all trades` style endpoint if I can help it. My only exception to this would be if we switched over to using a GraphQL API.
